src/components/ColorPicker/BrandColorPicker.tsx

```diff
@@ -81,18 +84,22 @@ export const BrandColorPicker: FC<Props> = ({ palettes: defaultPalettes = [], cu
                                   ])}
                               >
                                   {colors.map((color) => (
-                                      <li key={color.hex} data-test-id="brand-color">
+                                      <li key={color.name} data-test-id="brand-color">
                                           <button
                                               className="tw-flex tw-overflow-hidden tw-w-full"
-                                              onClick={() => onSelect(toColor(color, color))}
+                                              onClick={() => onSelect(color)}
                                           >
                                               <span
-                                                  className="tw-h-8 tw-w-8 tw-mr-2 tw-rounded tw-flex tw-items-center tw-justify-center tw-text-white"
-                                                  style={{ background: color.hex }}
+                                                  className={`tw-h-8 tw-w-8 tw-mr-2 tw-rounded tw-flex tw-items-center tw-justify-center ${
+                                                      isColorLight(color) ? "tw-text-black" : "tw-text-white"
+                                                  }`}
+                                                  style={{ background: tinycolor(color).toRgbString() }}
                                               >
-                                                  {color.hex === currentColor.hex && (
-                                                      <IconCheck size={IconSize.Size20} />
-                                                  )}
+                                                  {color.r === currentColor.r &&
+                                                      color.g === currentColor.g &&
+                                                      color.b === currentColor.b && (
```

Comment:
Probably we should consider `a` too here

Reply:
Good catch, added 👍🏻